perf(linter/plugins): remove `Arc`s

overlookmotel opened the pull request:
Remove 2 levels of `Arc`s from `ExternalLinter`.

1. Make wrapped callback functions `Box<dyn Fn>` instead of `Arc<dyn Fn>`.
2. Avoid wrapping `ThreadsafeFunction`s in `Arc`. The closures take ownership of them anyway. In `wrap_load_plugin`, make the inner closure borrow the `ThreadsafeFunction` from the outer closure.

The first of these (and possibly the 2nd too) was prevented previously by `#[derive(Clone)]` on `Linter`. `Linter` doesn't appear to need to be cloned, so remove that derive.

camc314 commented:
> @camc314 Can I check it's OK to remove `#[derive(Clone)]` from `Linter`? Doesn't seem to break anything, and I'm unclear why you'd want to clone it anyway, but there may be something I'm missing...

don't think it's needed

overlookmotel committed:
perf(linter/plugins): remove `Arc`s (#15431)

Remove 2 levels of `Arc`s from `ExternalLinter`.

1. Make wrapped callback functions `Box<dyn Fn>` instead of `Arc<dyn Fn>`.
2. Avoid wrapping `ThreadsafeFunction`s in `Arc`. The closures take ownership of them anyway. In `wrap_load_plugin`, make the inner closure borrow the `ThreadsafeFunction` from the outer closure.

The first of these (and possibly the 2nd too) was prevented previously by `#[derive(Clone)]` on `Linter`. `Linter` doesn't appear to need to be cloned, so remove that derive.

overlookmotel commented on apps/oxlint/src/js_plugins/external_linter.rs:
```diff
@@ -36,10 +36,9 @@ pub fn create_external_linter(
-        tokio::task::block_in_place(move || {
+    Box::new(move |plugin_path, package_name| {
+        let cb = &cb;
+        tokio::task::block_in_place(|| {
```
Completely wrong. The current code compiles. Whereas adding `move` to the inner closure causes a compilation error. `cb.call_async(...)` inside the inner closure only requires a `&cb` reference.
